src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -32,21 +32,24 @@ class EntityUserProvider implements UserProviderInterface
     private $classOrAlias;
     private $class;
     private $property;
+    private $repository;
 
-    public function __construct(ManagerRegistry $registry, $classOrAlias, $property = null, $managerName = null)
+    public function __construct(ManagerRegistry $registry, $classOrAlias, $property = null, $managerName = null, $repository = null)
     {
         $this->registry = $registry;
         $this->managerName = $managerName;
         $this->classOrAlias = $classOrAlias;
         $this->property = $property;
+        $this->repository = $repository;
     }
 
     /**
      * {@inheritdoc}
      */
     public function loadUserByUsername($username)
     {
-        $repository = $this->getRepository();
+        $repository = (null !== $this->repository) ? $this->repository : $this->getRepository();
```

Comment:
i suggest to  handle the custom repo  in `getRepository()` as such

Reply:
Good one, I'm doing that a rebasing the PR. Thanks!